src/python/foglamp/core/server.py

```diff
@@ -30,6 +51,300 @@ class Server:
     scheduler = None
     """ foglamp.core.Scheduler """
 
+    logging_configured = False
+    """Set to true when it's safe to use logging"""
+
+    # TODO: Discuss if this needs to be pre-decided and fixed
+    _MANAGEMENT_API_PORT = 8081
+
+    # TODO: Set below storage ports to None after Storage layer is fixed to accept these values via command line arguments
+    _STORAGE_PORT = 8080
+    _STORAGE_MANAGEMENT_PORT = 1081
+
+    # TODO: Fix a TODO in _start() below to accept below via discovery
+    _RESTAPI_PORT = 8082
+
+    @staticmethod
+    def request_available_port():
+        s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        s.bind(('localhost', 0))
+        addr, port = s.getsockname()
+        s.close()
```

Comment:
A service (web, storage, etc.) must directly bind to and use a port. With this method, we request an open port, close it, and return the port number. Then we pass this as a config to a service to bind to and use. However, by the time the service uses the port, the port may have been claimed by another service or process.